crates/owl-ms-language-server/src/catalog.rs

```diff
@@ -0,0 +1,49 @@
+use std::fs::read_to_string;
+
+use itertools::Itertools;
+use log::info;
+use quick_xml::de::from_str;
+use serde::{Deserialize, Serialize};
+use tower_lsp::lsp_types::Url;
+use walkdir::WalkDir;
+
+#[derive(Deserialize, Serialize, Debug, Clone)]
+#[serde(rename_all = "kebab-case")]
+pub struct Catalog {
```

Comment:
Do you intend to support more complex catalog files as well? The full spec is [here](https://groups.oasis-open.org/higherlogic/ws/public/download/14810/xml-catalogs.pdf). But what I've seen most often are the use of groups e.g. [in UBERON](https://github.com/obophenotype/uberon/blob/master/subsets/catalog-v001.xml) or [in OEO](https://github.com/OpenEnergyPlatform/ontology/blob/dev/src/ontology/edits/catalog-v001.xml)

Reply:
Ohh thanks. I will look into it.

Reply:
Do you want to do this in this PR or a later one? Otherwise I think this can be merged.

Reply:
A later one. I have done some stuff already but felt like merging this PR now. The follow-up is #34